File: internal/model/stats.go
package model

import "time"

// Stats holds per-profile aggregate progress shown on the menu and stats screens.
type Stats struct {
	Streak        int
	BestStreak    int
	LastStudiedAt time.Time // zero value means the profile has never studied
	XP            int       // cumulative experience points earned across all activity
}

File: core/src/model/stats.rs
use chrono::{DateTime, Utc};

/// Per-profile aggregate progress shown on the menu and stats screens.
#[derive(Clone, Debug, Default, PartialEq)]
pub struct Stats {
    pub streak: i64,
    pub best_streak: i64,
    /// `None` means the profile has never studied.
    pub last_studied_at: Option<DateTime<Utc>>,
    /// Cumulative experience points earned across all activity.
    pub xp: i64,
}
